lib/cocina/models/builders/title_builder.rb

```diff
@@ -149,68 +184,73 @@ def other_title(titles)
-              non_sort_value = "#{value}#{non_sorting_padding(title, value)}"
+              padding = non_sorting_padding(cocina_title, value)
+              result = add_non_sorting_value(result, value, padding)
             when 'part name', 'part number'
-              part_name_number = part_name_number(title.structuredValue) if part_name_number.blank?
-            when 'main title', 'title'
-              main_title = value
-            when 'subtitle'
-              # combine multiple subtitles into a single string
-              subtitle = if !add_punctuation?
-                           if subtitle.present?
-                             [subtitle, value].join(' ')
-                           else
-                             value
-                           end
-                         elsif subtitle.present?
-                           # subtitle is preceded by space colon space, unless it is at the beginning of the title string
-                           "#{subtitle.sub(/[. :]+$/, '')} : #{value.sub(/^:/, '').strip}"
+              if part_name_number.blank?
+                part_name_number = part_name_number(cocina_title.structuredValue)
+                result = if !add_punctuation?
+                           [result, part_name_number].join(' ')
+                         elsif result.present?
+                           "#{result.sub(/[ .,]*$/, '')}. #{part_name_number}. "
                          else
-                           value.sub(/^:/, '').strip
+                           "#{part_name_number}. "
                          end
+              end
+            when 'main title', 'title'
+              result = if add_punctuation? || result.ends_with?(' ') || result.ends_with?('-') || result.ends_with?('\'')
+                         [result, value].join
+                       else
+                         [remove_trailing_punctuation(result), remove_trailing_punctuation(value)].select(&:presence).join(' ')
+                       end
+            when 'subtitle'
+              result = if !add_punctuation?
+                         [result, value].select(&:presence).join(' ')
+                       elsif result.present?
+                         # subtitle is preceded by space colon space, unless it is at the beginning of the title string
+                         "#{result.sub(/[. :]+$/, '')} : #{value.sub(/^:/, '').strip}"
+                       else
+                         result = value.sub(/^:/, '').strip
+                       end
             end
           end
 
-          # combine the parts into a single title string
-          if add_punctuation?
-            combine_with_punctuation(non_sort_value: non_sort_value, main_title: main_title, subtitle: subtitle,
-                                     part_name_number: part_name_number)
-          else
-            ["#{non_sort_value}#{main_title}", subtitle, part_name_number].select(&:presence).join(' ')
-          end
```

Comment:
this was me being too clever about putting deconstructed title strings back together